src/scope-locals.ts

```diff
@@ -79,6 +79,10 @@ export class ScopeLocals {
     return ['this', 'globalThis'].includes(jsName) || jsPath.scope.getBinding(jsName);
   }
 
+  #isImplicitScope(name: string) {
```

Comment:
the fix so far -- step next is to throw a hard error if `this` is referenced outside of a class or function